spatialdata/_core/_spatialdata.py

```diff
@@ -0,0 +1,262 @@
+from __future__ import annotations
+
+from functools import singledispatch
+from types import MappingProxyType
+from typing import Any, Mapping, Optional, Union
+
+import numpy as np
+import zarr
+from anndata import AnnData
+from dask.array.core import Array as DaskArray
+from geopandas import GeoDataFrame
+from multiscale_spatial_image.multiscale_spatial_image import MultiscaleSpatialImage
+from ome_zarr.io import parse_url
+from spatial_image import SpatialImage
+
+from spatialdata._core.models import (
+    Image2DModel,
+    Image3DModel,
+    Label2DModel,
+    Label3DModel,
+    PointModel,
+    PolygonModel,
+    ShapeModel,
+    TableModel,
+)
+from spatialdata._io.write import (
+    write_image,
+    write_labels,
+    write_points,
+    write_polygons,
+    write_shapes,
+    write_table,
+)
+
+# schema for elements
+Label2d_s = Label2DModel()
+Label3D_s = Label3DModel()
+Image2D_s = Image2DModel()
+Image3D_s = Image3DModel()
+Polygon_s = PolygonModel
+Point_s = PointModel()
+Shape_s = ShapeModel()
+Table_s = TableModel()
+
+
+class SpatialData:
+    """Spatial data structure."""
+
+    images: Mapping[str, Union[SpatialImage, MultiscaleSpatialImage]] = MappingProxyType({})
+    labels: Mapping[str, Union[SpatialImage, MultiscaleSpatialImage]] = MappingProxyType({})
+    points: Mapping[str, AnnData] = MappingProxyType({})
+    polygons: Mapping[str, GeoDataFrame] = MappingProxyType({})
+    shapes: Mapping[str, AnnData] = MappingProxyType({})
+    _table: Optional[AnnData] = None
+
+    def __init__(
+        self,
+        images: Mapping[str, Any] = MappingProxyType({}),
+        labels: Mapping[str, Any] = MappingProxyType({}),
+        points: Mapping[str, Any] = MappingProxyType({}),
+        polygons: Mapping[str, Any] = MappingProxyType({}),
+        shapes: Mapping[str, Any] = MappingProxyType({}),
+        table: Optional[AnnData] = None,
+        **kwargs: Any,
+    ) -> None:
+
+        if images is not None:
+            self.images = {}
```

Comment:
I think this is a bit confusing because the default element property values and `__init__` parameters values are `MappingProxyType`s while then end up getting initialized as regular dictionaries. The `MappingProxyType` would be readonly while the dictionary is mutable.

What is the intended behavior? Should we make the elements properties without settings so that they can't be directly mutated?